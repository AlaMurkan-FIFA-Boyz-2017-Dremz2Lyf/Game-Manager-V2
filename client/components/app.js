// Dependancies
import React, { Component } from 'react';
import { Link } from 'react-router';
import {
  PageHeader,
  Grid
} from 'react-bootstrap';


class App extends Component {

  render() {
    return (
      <div className="home">
        <Link to='/'><PageHeader>Game Manager</PageHeader></Link>
        <Grid>
          {this.props.children}
        </Grid>
      </div>
    );
  }
}



export default App;
