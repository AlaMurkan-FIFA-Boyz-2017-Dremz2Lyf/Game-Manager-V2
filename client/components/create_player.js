import React, { Component } from 'react';
import { Modal, ListGroupItem, Button } from 'react-bootstrap';
import { connect } from 'react-redux';

import PlayerForm from './player_form';
import { create } from '../actions/index';

export class CreatePlayer extends Component {
  constructor() {
    super();

    this.state = {
      showModal: false
    };
  }

  open() {
    this.setState({
      showModal: true
    });
  }

  close() {
    this.setState({
      showModal: false
    });
  }

  handleSubmit(values) {
    this.props.create('players', values);
    this.setState({
      showModal: false
    });
  }

  render() {

    return (
      <div>
        <ListGroupItem onClick={this.open.bind(this)}>
          Couldn't find who you were looking for? Click here to add a new player
        </ListGroupItem>
        <Modal show={this.state.showModal} onHide={this.close.bind(this)}>
          <Modal.Header>
            <Modal.Title>Make a new Player</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <PlayerForm onSubmit={this.handleSubmit.bind(this)} />
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={this.close.bind(this)}>Cancel</Button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  }
}

export default connect(null, { create })(CreatePlayer);
