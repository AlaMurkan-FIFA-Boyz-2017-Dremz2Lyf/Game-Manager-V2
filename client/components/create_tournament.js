import React, { Component } from 'react';
import { connect } from 'react-redux';

import {
  Row,
  Col,
  ListGroupItem,
  FormControl,
  FormGroup,
  ControlLabel,
  InputGroup,
  DropdownButton,
  MenuItem,
  Button
} from 'react-bootstrap';

import PlayerList from './player_list';
import CreatePlayer from './create_player.js';
import PlayerForm from './player_form';

import { create } from '../actions/index';

export class CreateTournament extends Component {
  constructor() {
    super();

    this.state = {
      playerSearch: '',
      name: '',
      rounds: '1 Round',
      added: []
    };
  }
  // All of the event handlers
  handleRounds(rounds) {
    this.setState({
      rounds: rounds
    });
  }

  handleSearch(e) {
    this.setState({
      playerSearch: e.target.value
    });
  }

  handlerName(e) {
    this.setState({
      name: e.target.value
    });
  }

  // handles the submittion of the tournament, and creates it with a post to the database.
  handleSubmit(e) {
    e.preventDefault();
    let { rounds, added, name } = this.state;
    rounds = rounds.slice(0, 1);
    this.props.create('tournaments', { name, rounds, added });
  }

  // added takes boolean dictating if the list renders the players added to the tournament (true) or all the others (false).
  added(includeAdded) {
    let { players = {} } = this.props;
    let { added } = this.state;

    return Object.keys(players).filter(id =>
      added.includes(id) === includeAdded
    ).map(id => players[id]);
  }

  // move will check the local state to see if the id is in the added Array
    // if it is, remove it. Otherwise we add it.
  move(id) {
    let { added } = this.state;
    let newAdded = added.indexOf(id) !== -1 ? (
      added.filter((addedId) => addedId !== id)
    ) : (
      [...added, id]
    );
    this.setState({
      added: newAdded
    });
  }

  renderOptions(number) {
    return [<MenuItem onClick={this.handleRounds.bind(this, '1 Round')} key={1}>1 Round</MenuItem>].concat([...Array(number).keys()].slice(1)
      .filter(num => num % 2 === 0 )
      .map(num => (<MenuItem onClick={this.handleRounds.bind(this, `${num} Rounds`)} key={num}>{num} Rounds</MenuItem>)));
  }



  render() {
    let { rounds, name, playerSearch } = this.state;

    return (
      <div>
        <Row>
          <Col xs={12}>
          <form onSubmit={this.handleSubmit.bind(this)}>
            <FormGroup>
              <ControlLabel>Tournament Name</ControlLabel>
                <InputGroup>

                <InputGroup.Button>
                  <Button type='submit'>Go!</Button>
                </InputGroup.Button>
                <FormControl
                  name='name'
                  type='text'
                  placeholder='Tournament Name'
                  onChange={this.handlerName.bind(this)}
                  value={name}
                />
                <DropdownButton
                  componentClass={InputGroup.Button}
                  title={rounds}
                  id='tournament_rounds'
                >
                  {this.renderOptions(13)}
                </DropdownButton>
                </InputGroup>
              </FormGroup>
            </form>
          </Col>
        </Row>
        <Row>
          <Col xs={12}>
          <ControlLabel>Add Players!</ControlLabel>
          <div>
            <FormControl
              onChange={this.handleSearch.bind(this)}
              value={playerSearch}
              name='playerSearch'
              type='text'
              placeholder='Who do you want to play with?'
            />
          </div>
          </Col>
        </Row>
        <Row>
          <Col xs={6}>
            <PlayerList
              name='all-players'
              players={this.added(false)}
              move={this.move.bind(this)}
              searchValue={playerSearch}
              all={true}
            />
          </Col>
          <Col xs={6}>
            <PlayerList
              name='added-players'
              players={this.added(true)}
              move={this.move.bind(this)}
              searchValue={''}
              all={false}
            />
          </Col>
        </Row>
      </div>
    );
  }
}

export const mapStateToProps = ({data}) => {
  return {
    players: data.players
  };
};

export default connect(mapStateToProps, { create })(CreateTournament);
