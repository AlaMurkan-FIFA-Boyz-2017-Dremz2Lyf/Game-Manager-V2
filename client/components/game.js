import React, { Component } from 'react';
import { connect } from 'react-redux';

import { update } from '../actions';

import {
  Modal,
  ListGroupItem,
  Button,
  Row,
  Col
} from 'react-bootstrap';

import GameDetails from './game_details';
import GameForm from './game_form';

export class Game extends Component {
  constructor() {
    super();
    this.state = {
      showModal: false
    };
  }

  open() {
    this.setState({
      showModal: true
    });
  }

  close() {
    this.setState({
      showModal: false
    });
  }

  handleSubmit(values) {
    // Pull the game, the tournaments, and the update action creator from props.
    let { game, tournaments, update } = this.props;

    // then grab the id, tournament id and p1Score from the game.
    let { id, tournament, p1Score} = game;

    // lastly, grab the gamesPlayed from the tournament corresponding to this game.
    let { gamesPlayed } = tournaments[tournament];

    // Because both scores are required to update a game, we can check if the game has been played with just one of the scores.
    if (p1Score === null) {
      // If this is the first time this game is being updated, we increment the game count, and update the tournament.
      ++gamesPlayed;
      update('tournaments', {id: tournament, gamesPlayed});
    }

    // Update the game in the database with the values from the form and a status of finished.
    update('games', {id, ...values, status: 'finished'});
    this.setState({
      showModal: false
    });
  }

  render() {
    let { game } = this.props;
    let { players } = this.props;

    return (
      <ListGroupItem onClick={this.open.bind(this)}>
        <GameDetails
          game={game}
          player1={players[game.p1]}
          player2={players[game.p2]}
        />
        <Modal show={this.state.showModal} onHide={this.close.bind(this)}>
          <Modal.Header>
            <Modal.Title>Finish this Game!</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Row>
              <Col xs={3}>
              </Col>
              <Col xs={2}>
                <h4>{players[game.p1].username}</h4>
              </Col>
              <Col xs={2}>
                <h4>{players[game.p2].username}</h4>
              </Col>
              <Col xs={5}>
              </Col>
            </Row>
            <Row>
              <Col xs={12}>
                <GameForm
                  onSubmit={this.handleSubmit.bind(this)}
                  initialValues={game}
                />
              </Col>
            </Row>
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={this.close.bind(this)}>Cancel</Button>
          </Modal.Footer>
        </Modal>
      </ListGroupItem>
    );
  }
}

Game.propTypes = {
  game: React.PropTypes.object.isRequired,
  players: React.PropTypes.object.isRequired
};

export const mapStateToProps = ({ data }) => ({
  players: data.players,
  tournaments: data.tournaments
});

export default connect(mapStateToProps, { update })(Game);
