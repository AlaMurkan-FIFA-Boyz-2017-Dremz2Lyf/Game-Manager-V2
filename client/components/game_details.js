import React from 'react';

import { Row, Col, ProgressBar } from 'react-bootstrap';

const GameDetails = ({ game, player1, player2 }) => (
  <Row className='inner-text hover'>
    <Col xs={5}>
      <span className='player1-name'>{player1.username}</span>
      <span className='player1-score'>{game.p1Score}</span>
    </Col>
    <Col xs={2} className='vs'>
      vs.
    </Col>
    <Col xs={5}>
      <span className='player2-score'>{game.p2Score}</span>
      <span className='player2-name'>{player2.username}</span>
    </Col>
  </Row>
);


GameDetails.propTypes = {
  game: React.PropTypes.object.isRequired,
  player1: React.PropTypes.object.isRequired,
  player2: React.PropTypes.object.isRequired
};

export default GameDetails;
