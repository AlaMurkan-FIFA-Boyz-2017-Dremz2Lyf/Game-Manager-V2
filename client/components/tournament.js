import React from 'react';
import { ProgressBar, Col, Row, ListGroupItem } from 'react-bootstrap';

import { percentPlayed } from '../utilities';

export const Tournament = ({ tournament = {} }) => {
  let { name, createdAt, gamesPlayed, totalGames } = tournament;
  return (
    <ListGroupItem className="tournament">
      <Row>
        <Col xs={7} className="name">
          <div>{name}</div>
        </Col>
        <Col xs={5} className="date">
          <span>{createdAt}</span>
        </Col>
      </Row>
      <Row>
        <Col xs={12}>
          <ProgressBar active now={percentPlayed(gamesPlayed, totalGames)}></ProgressBar>
        </Col>
      </Row>
    </ListGroupItem>
  );
};
