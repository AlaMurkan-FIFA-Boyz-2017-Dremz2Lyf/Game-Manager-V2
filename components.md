 Component | State | Actions
---|---|---
 Home | None? | None?
 AllTimeStats | {currentList: ongoing/finished/new} | None?
