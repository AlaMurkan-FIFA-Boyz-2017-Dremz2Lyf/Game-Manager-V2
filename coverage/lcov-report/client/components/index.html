<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> client/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>95/95</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>25/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>51/51</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>91/91</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="create_player.js"><a href="create_player.js.html">create_player.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="create_tournament.js"><a href="create_tournament.js.html">create_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	</tr>

<tr>
	<td class="file high" data-value="game.js"><a href="game.js.html">game.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="games.js"><a href="games.js.html">games.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="home.js"><a href="home.js.html">home.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="play_tournament.js"><a href="play_tournament.js.html">play_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="player_form.js"><a href="player_form.js.html">player_form.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list.js"><a href="player_list.js.html">player_list.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list_item.js"><a href="player_list_item.js.html">player_list_item.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="tournament.js"><a href="tournament.js.html">tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="tournaments.js"><a href="tournaments.js.html">tournaments.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Mar 10 2017 17:46:58 GMT-0600 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
