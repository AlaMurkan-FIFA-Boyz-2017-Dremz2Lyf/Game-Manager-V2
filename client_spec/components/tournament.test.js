import React from 'react';
import { Tournament } from '../../client/components/tournament';


describe('Tournament component', () => {
  let wrapper = shallow(<Tournament />);

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
