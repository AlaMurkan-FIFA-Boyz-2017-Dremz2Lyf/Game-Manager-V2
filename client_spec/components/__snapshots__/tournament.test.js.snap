exports[`Tournament component should match the snapshot 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  className="tournament"
  listItem={false}>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      className="name"
      componentClass="div"
      xs={7}>
      <div />
    </Col>
    <Col
      bsClass="col"
      className="date"
      componentClass="div"
      xs={5}>
      <span />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <ProgressBar
        active={true}
        bsClass="progress-bar"
        isChild={false}
        max={100}
        min={0}
        now={NaN}
        srOnly={false}
        striped={false} />
    </Col>
  </Row>
</ListGroupItem>
`;
